pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercuryOrderDao.java

```diff
@@ -121,69 +132,74 @@ public static void updateOrderStatus(BaseMercuryStatusMessage baseMercuryStatusM
-                        log.info("found related info about barcode " + mercuryOrderDto.getBarcode());
-                        map.put(mercuryOrderDto.getBarcode(), mercuryOrderDto);
-                    }
-                } catch (Exception e) {
-                    throw new RuntimeException("Error getting possible mercury orders for participant " + ddpParticipantId, e);
-                }
-            } catch (SQLException ex) {
+                getKitsFromStatement(stmt, map);
+            } catch (Exception ex) {
                 dbVals.resultException = ex;
             }
 
             return dbVals;
         });
 
         if (results.resultException != null) {
-            throw new RuntimeException("Error getting possible mercury orders for participant " + ddpParticipantId,
+            throw new DsmInternalError(String.format(errorMessage, ddpParticipantId),
                     results.resultException);
         }
 
         results = inTransaction((conn) -> {
             SimpleResult dbVals = new SimpleResult();
             try (PreparedStatement stmt = conn.prepareStatement(SQL_GET_KIT_FROM_BARCODE_SM_ID)) {
                 stmt.setString(1, ddpParticipantId);
-                try (ResultSet rs = stmt.executeQuery()) {
-                    while (rs.next()) {
-                        MercuryOrderDto mercuryOrderDto = new MercuryOrderDto(rs.getString(DBConstants.DDP_PARTICIPANT_ID),
-                                rs.getString(DBConstants.MERCURY_ORDER_CREATOR), rs.getString(DBConstants.SM_ID_VALUE),
-                                rs.getInt("ktype." + DBConstants.KIT_TYPE_ID), rs.getInt(DBConstants.DDP_INSTANCE_ID),
-                                rs.getLong(DBConstants.TISSUE_ID), null);
-                        log.info("found related info about barcode " + mercuryOrderDto.getBarcode());
-                        map.put(mercuryOrderDto.getBarcode(), mercuryOrderDto);
-                    }
-                } catch (Exception e) {
-                    throw new RuntimeException("Error getting possible mercury orders for participant " + ddpParticipantId, e);
-                }
+                getBarcodesFromResultSet(stmt, map, ddpParticipantId);
             } catch (SQLException ex) {
                 dbVals.resultException = ex;
             }
-
             return dbVals;
         });
 
         if (results.resultException != null) {
-            throw new RuntimeException("Error getting the mercury orders", results.resultException);
+            throw new DsmInternalError("Error getting the mercury orders", results.resultException);
         }
         log.info(String.format("Found %d possible barcodes for participant %s", map.size(), ddpParticipantId));
         return map;
     }
 
+    private void getBarcodesFromResultSet(PreparedStatement stmt, HashMap<String, MercuryOrderDto> map, String ddpParticipantId) throws DsmInternalError {
```

Comment:
Why did you break these out of their prior methods? Was the linter complaining about something? With the change it's harder to associate the query with the statement result. Perhaps another way to do this is to break out just the part that creates the `MercuryOrderDto`? 

I'm also a bit concerned about changing anything unneeded unless there are automated tests covering at least the main code path. Do we have those for these?

Reply:
Yeah I tried to reduce linter errors, sonar was really not happy with nested tries, however, I really hate these changes myself. Makes it less readable, and I wasn't sure what to put in for java docs. I think I'll just move them back and just try to fix it in the tries